scipy/stats/_continuous_distns.py

```diff
@@ -5206,6 +5206,92 @@ def _entropy(self, beta):
 halfgennorm = halfgennorm_gen(a=0, name='halfgennorm')
 
 
+class crystalball_gen(rv_continuous):
+    """
+    Crystalball distribution
+    
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for `crystalball` is::
+
+                                            --
+                                            | exp(-x**2 / 2),  for x > -alpha
+        crystalball.pdf(x, alpha, n) =  N * |
+                                            | A * (B - x)**(-n), for x <= -alpha
+                                            --
+        where:
+                                   A = (n / |alpha|)**n * exp(-alpha**2 / 2)
+                                   B = n / |alpha| - |alpha|
+                                   and N is a normalisation constant.
+
+    `crystalball` takes ``alpha`` and ``n`` as shape parameters.
+    ``alpha`` defines the point where the pdf changes from a power-law to a gaussian distribution
+    ``n`` is power of the power-law tail.
+
+    References
+    ----------
+    .. [1] "Crystal Ball Function",
+           https://en.wikipedia.org/wiki/Crystal_Ball_function
+
+    %(after_notes)s
+    
+    .. versionadded: 0.19.0
+
+    %(example)s
+    """
+    def _pdf(self, x, alpha, n):
+        """
+        Return PDF of the crystalball function.
+        """
+        N = 1.0 / (n/alpha / (n-1) * np.exp(-alpha**2 / 2.0) + _norm_pdf_C * _norm_cdf(alpha))
+        return N * _lazywhere(np.atleast_1d(x > -alpha), (x, alpha, n),
+                              f = lambda x, alpha, n: np.exp(-x**2 / 2),
+                              f2 = lambda x, alpha, n: (n/alpha)**n * np.exp(-alpha**2 / 2.0) * (n/alpha - alpha - x)**(-n))
```

Comment:
I refactored the lazywhere and introduced a right-hand-side (rhs) and left-hand-side (lhs)
lambda function which is passed to f and f2 of _lazywhere.
The lines are now shorter.